src/vs/workbench/contrib/terminal/browser/terminalInstance.ts

```diff
@@ -917,19 +917,36 @@ export class TerminalInstance extends Disposable implements ITerminalInstance {
 
 	async runCommand(commandLine: string, shouldExecute: boolean): Promise<void> {
 		let commandDetection = this.capabilities.get(TerminalCapability.CommandDetection);
+		const timeOutValue = this._configurationService.getValue(TerminalSettingId.ShellIntegrationTimeout) as number | undefined;
+		let timeoutMs: number;
+
+		if (timeOutValue === undefined || typeof timeOutValue !== 'number' || timeOutValue < 0) {
+			const siEnabled = this._configurationService.getValue(TerminalSettingId.ShellIntegrationEnabled) === true;
+			// Discuss: This default from copilot chat setting feels pretty slow..
```

Comment:
It's slow, but only if things don't work and only for the first command. We need to check the time against when `processReady` happens